### ISSUE TITLE
Java 13 Warning: Unsupported class file major version 57

### ISSUE THREAD

**Issue by adamu** [mockk/mockk#397]
Mockk produces a warning when running on Java 13. ~It still seems to work in my small test, but I'm not sure if there are any other implications.~ edit: I've discovered it fails on Static mocks. Added details below.

### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

Mockk works with JDK 13 without warnings.

### Current Behavior

```
WARNING: Failed to transform class java/lang/Object
java.lang.IllegalArgumentException: Unsupported class file major version 57
```

### Failure Information (for bugs)

I created a simple test file (below), which runs fine on JDK 11, but produces the above warning on JDK 13.

#### Context

* MockK version: 1.9.3
* OS: OSX 10.14.6
* Kotlin version: 1.3.61
* JDK version: 11.0.5 (no warning) / 13.0.1 (warning)
* JUnit version: N/A
* Type of test: Simple test file

#### Stack trace

```
Not mockk OK: true
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
Dec 03, 2019 6:52:51 PM io.mockk.impl.log.JULLogger warn
WARNING: Failed to transform class java/lang/Object
java.lang.IllegalArgumentException: Unsupported class file major version 57
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:184)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:166)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:152)
	at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:82)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3390)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:1930)
	at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:217)
	at net.bytebuddy.dynamic.scaffold.inline.AbstractInliningDynamicTypeBuilder.make(AbstractInliningDynamicTypeBuilder.java:120)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3396)
	at io.mockk.proxy.jvm.transformation.InliningClassTransformer.transform(InliningClassTransformer.kt:77)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:167)
	at io.mockk.proxy.jvm.transformation.JvmInlineInstrumentation.retransform(JvmInlineInstrumentation.kt:28)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:19)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:6)
	at io.mockk.proxy.common.transformation.ClassTransformationSpecMap.applyTransformation(ClassTransformationSpecMap.kt:41)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation.execute(RetransformInlineInstrumnetation.kt:16)
	at io.mockk.proxy.jvm.ProxyMaker.inline(ProxyMaker.kt:88)
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:30)
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:34)
	at io.mockk.impl.instantiation.AbstractMockFactory.newProxy$default(AbstractMockFactory.kt:29)
	at io.mockk.impl.instantiation.AbstractMockFactory.mockk(AbstractMockFactory.kt:59)
	at MockkTestKt.main(MockkTest.kt:29)
	at MockkTestKt.main(MockkTest.kt)

Mockk OK: true
Done.

Process finished with exit code 0
```

#### Minimal reproducible code (the gist of this issue)

```
// -----------------------[ GRADLE DEFINITIONS ] -----------------------
dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib:1.3.61"
    implementation "io.mockk:mockk:1.9.3"
}
```

```kotlin
class MockkMe(val foo: String)

fun main() {
    val notMock = MockkMe("foo")
    println("Not mockk OK: " + (notMock.foo == "foo"))

    val mockk: MockkMe = io.mockk.mockk()
    io.mockk.every { mockk.foo } returns "bar"
    println("Mockk OK: " + (mockk.foo == "bar"))

    println("Done.")
}
```

## Version with static mock and output:

```kotlin
import io.mockk.every
import io.mockk.mockkStatic
import java.time.Instant

fun main() {
    val epoch = Instant.EPOCH
    mockkStatic(Instant::class)
    every { Instant.now() } returns epoch
    println("Static Mockk OK: " + (Instant.now() == epoch))
}
```

```
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
Dec 03, 2019 7:08:47 PM io.mockk.impl.log.JULLogger warn
WARNING: Failed to transform class java/time/Instant
java.lang.IllegalArgumentException: Unsupported class file major version 57
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:184)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:166)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:152)
	at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:82)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3390)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:1930)
	at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:217)
	at net.bytebuddy.dynamic.scaffold.inline.AbstractInliningDynamicTypeBuilder.make(AbstractInliningDynamicTypeBuilder.java:120)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3396)
	at io.mockk.proxy.jvm.transformation.InliningClassTransformer.transform(InliningClassTransformer.kt:77)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:167)
	at io.mockk.proxy.jvm.transformation.JvmInlineInstrumentation.retransform(JvmInlineInstrumentation.kt:28)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:19)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation$execute$1.invoke(RetransformInlineInstrumnetation.kt:6)
	at io.mockk.proxy.common.transformation.ClassTransformationSpecMap.applyTransformation(ClassTransformationSpecMap.kt:41)
	at io.mockk.proxy.common.transformation.RetransformInlineInstrumnetation.execute(RetransformInlineInstrumnetation.kt:16)
	at io.mockk.proxy.jvm.StaticProxyMaker.staticProxy(StaticProxyMaker.kt:33)
	at io.mockk.impl.instantiation.JvmStaticMockFactory.staticMockk(JvmStaticMockFactory.kt:39)
	at MockkTestKt.main(MockkTest.kt:20)
	at MockkTestKt.main(MockkTest.kt)

Exception in thread "main" io.mockk.MockKException: Missing calls inside every { ... } block.
	at io.mockk.impl.recording.states.StubbingState.checkMissingCalls(StubbingState.kt:14)
	at io.mockk.impl.recording.states.StubbingState.recordingDone(StubbingState.kt:8)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
	at io.mockk.impl.eval.EveryBlockEvaluator.every(EveryBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalEvery(API.kt:92)
	at io.mockk.MockKKt.every(MockK.kt:104)
	at MockkTestKt.main(MockkTest.kt:8)
	at MockkTestKt.main(MockkTest.kt)

Process finished with exit code 1
```

Above test also works fine on Java 11.

**Comment by igorakkerman** [comment]
Since the error appears to be caused by Byte Buddy, I added a Gradle dependency to its latest version. This workaround solves the issue for me, running on OpenJDK 13.0.1:
`testRuntimeOnly("net.bytebuddy:byte-buddy:1.10.6")`
MockK currently has a dependency on Byte Buddy version 1.9.10, upgrading might help.

**Comment by utybo** [comment]
It looks like this was fixed in the latest release of MockK (1.10.0)

**Comment by schmidtDTN** [comment]
I'm seeing this same issue with any version of MockK up to the latest version (1.13.7) while upgrading to JDK 17 - the initial symptom is slightly different but I'm getting the following:
```
[Test worker] WARN io.mockk.proxy.jvm.transformation.InliningClassTransformer - Failed to transform class X 
java.lang.IllegalArgumentException: Unsupported class file major version 61
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:189)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:170)
	at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:156)
	at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:86)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3889)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2166)
	...
```
```
I fixed it by adding the following line to my `build.gradle.kts`:
`testRuntimeOnly("net.bytebuddy:byte-buddy:1.10.21")`

Just putting this here for anyone else who encounters this issue!
